sdk/framework/openapi.go

```diff
@@ -558,6 +534,102 @@ func specialPathMatch(path string, specialPaths []string) bool {
 	return false
 }
 
+// constructOperationID joins the given inputs into a title case operation id,
+// which is also used as a prefix for request and response names.
+//
+// The OperationPrefix / OperationSuffix / Action found in display attributes
+// will be used, if provided. Otherwise, the function falls back to using the
+// path and the operation.
+//
+// Examples of generated operation identifiers:
+//   - KVv2Write
+//   - KVv2Read
+//   - GoogleCloudLogin
+//   - GoogleCloudWriteRole
+func constructOperationID(
+	path string,
+	pathAttributes *DisplayAttributes,
+	operation logical.Operation,
+	operationAttributes *DisplayAttributes,
+	defaultPrefix string,
+) string {
+	var (
+		prefix string
+		suffix string
+		verb   string
+	)
+
+	if operationAttributes != nil {
+		prefix = operationAttributes.OperationPrefix
+		suffix = operationAttributes.OperationSuffix
+		verb = operationAttributes.Action
+	}
+
+	if pathAttributes != nil {
+		if prefix == "" {
+			prefix = pathAttributes.OperationPrefix
+		}
+		if suffix == "" {
+			suffix = pathAttributes.OperationSuffix
+		}
+		if verb == "" {
+			verb = pathAttributes.Action
+		}
+	}
+
+	// disambiguate the suffix for the paths generated through `expandPattern`
```

Comment:
Is this actually needed? I see you have support for defining the suffix at both path and operation levels, so the suffix could be written in singular form on the path DisplayAttrs, and then overridden to plural form in the ListOperation DisplayAttrs? That would (I think?) allow the same end goal without needing to hardcode particular parameter names here.

Reply:
This was my first attempt at addressing the one to many ambiguity introduced by the `expandPattern` (https://github.com/hashicorp/vault/pull/18321#discussion_r1063235356).

I have since come up with a (hopefully) better solution in d3e16aaf967520a92cf20dd3090321603964683a. Let me know what you think :)